ledger/sandbox-on-x/src/main/scala/com/daml/ledger/sandbox/ReadServiceWithFeedSubscriber.scala

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml
+
+import ledger.api.health.{HealthStatus, Healthy}
+import ledger.configuration.{Configuration, LedgerId, LedgerInitialConditions, LedgerTimeModel}
+import ledger.offset.Offset
+import ledger.participant.state.v2.{ReadService, Update}
+import lf.data.Time.Timestamp
+import logging.{ContextualizedLogger, LoggingContext}
+import akka.NotUsed
+import akka.stream.Materializer
+import akka.stream.scaladsl.Source
+import org.reactivestreams.Subscriber
+import scala.util.chaining._
+
+trait ReadServiceWithFeedSubscriber extends ReadService {
+  def subscriber: Subscriber[(Offset, Update)]
+}
+
+object ReadServiceWithFeedSubscriber {
+  def apply(
+      ledgerId: LedgerId,
+      maxDedupSeconds: Int,
+  )(implicit
+      loggingContext: LoggingContext,
+      materializer: Materializer,
+  ) = new ReadServiceWithFeedSubscriberImpl(ledgerId, maxDedupSeconds)
+
+  class ReadServiceWithFeedSubscriberImpl private[ReadServiceWithFeedSubscriber] (
+      ledgerId: LedgerId,
+      maxDedupSeconds: Int,
+  )(implicit
+      loggingContext: LoggingContext,
+      materializer: Materializer,
+  ) extends ReadServiceWithFeedSubscriber {
+    private val logger = ContextualizedLogger.get(getClass)
+    private var stateUpdatesWasCalledAlready = false
+
+    logger.info("Starting Sandbox-on-X read service...")
+
+    val (subscriber: Subscriber[(Offset, Update)], stateUpdatesSource) =
+      Source.asSubscriber[(Offset, Update)].preMaterialize().tap { _ =>
```

Comment:
do we really need reactive streams integration (java) to achieve this? It seems to me a simple MergeHub.source would also work here, no? :thinking: 
(https://doc.akka.io/docs/akka/2.5.32/stream/stream-dynamic.html#using-the-mergehub)

Reply:
I was not settled on this. I initially went to reactive streams since it is simpler and dumber. But since you asked, changed to using MergeHub